accepted/future-releases/unified-collections/feature-specification.md

```diff
@@ -655,12 +655,12 @@ They are defined as:
     expression evaluating to a value of type `bool` and either:
 
     *   If the condition evaluates to `true`, then when the "then" element is a
```

Comment:
I'm not sure the "when" and "if" are necessary here.

I think this sentence boils, logically, down to:

```
constIfElement = constBoolCondition &&
  (
    (conditionTrue && constThenElement && potentiallyConstElseElement)
    ||
    (conditionFalse && constElseElement && potentiallyConstThenElement)
  )
)
```

So I think in these "either" blocks you have three predicates ANDed together, and it might be clearer to write it like that then as an "if when"

Even though that's valid too, it requires imperative instead of functional programming:

```
isValid = constBoolCondition;
if (isValid) {
  switch(conditionBlockValue) {
    case true: return constThenElement && potentiallyConstElseElement;
    case false: return constElseElement && potentiallyConstThenElement;
  }
}
return isValid;
```

Which approach implementors use is up to them, but I think functional programming in EnglishSpecLanguage++ is clearer than imperative programming in EnglishSpecLanguage++.

Unless I'm spouting nonsense here.

Reply:
Your formulation is about right, except that it doesn't handle the else clauses being absent. In particular, it should be clear that omitting the else clause does *not* mean that neither case matches and it's impossible for an ifElement to be constant. Also, I'm not really sure how to map a switch to English. :)

I think the change here is at least a marginal improvement.

Reply:
I actually agree this is a mess to read.  It currently says:

An A is B if C and either:
  if X then when Y and Z.

To which I scratch my head and say "if X then when Y and Z what?"

Can we just rewrite this to:
```
 ... and either:
      *  The condition evaluates to `true`; the "then" element is a constant expression; and the "else" element, if it exists, is a potentially constant element.
      *  Or the condition evaluates to false....
```

Reply:
Done.